tables.go

```diff
@@ -87,7 +94,7 @@ func (t *table) putAtInt(k int, v value) {
 	} else if v == nil {
 		delete(t.hash, float64(k))
 	} else {
-		t.hash[float64(k)] = v
+		t.addOrInsertInMap(float64(k), v)
```

Comment:
Could we not call this `addOrInsertHash` to keep the language similar to `delete(t.hash...)` not really picky here..